pkg/controller/jobs/job/job_controller_test.go

```diff
@@ -2336,13 +2336,14 @@ func TestReconciler(t *testing.T) {
 				Key:      "tolerationkey2",
 				Operator: corev1.TolerationOpExists,
 				Effect:   corev1.TaintEffectNoSchedule,
-			}).
+			}).NodeSelector("node-label1", "value").
```

Comment:
I don't think the test verifies currently that nodeSelectors don't matter for the equality check since the tolerations are changed. Maybe we can have a new test "the workload is not admitted and nodeSelectors change", but only changing the node selectors?

Reply:
I would be fine with that approach. But we should have still have at least one higher-level test (like this) which corresponds to the reported bug. Maybe just rename the test "the workload is not admitted, tolerations and nodeSelectors change".